Formula/teip.rb

```diff
@@ -0,0 +1,23 @@
+class Teip < Formula
+  desc "Masking tape to help commands \"do one thing well\""
+  homepage "https://github.com/greymd/teip"
+  url "https://github.com/greymd/teip/archive/v2.3.0.tar.gz"
+  sha256 "4c39466613f39d27fa22ae2a6309ac732ea94fdbc8711ecd4149fc1ecdfbaedf"
+  license "MIT"
+  head "https://github.com/greymd/teip.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--features", "oniguruma", *std_cargo_args
```

Comment:
Since you're enabling `oniguruma`, you need a `depends_on "oniguruma"` and make sure it's being picked up. [We don't like vendored dependencies](https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae) when they are available.

See [`zola.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/zola.rb) for an example. Specifically, you need to set the [`RUSTONIG_SYSTEM_LIBONIG`](https://github.com/rust-onig/rust-onig/blob/main/onig_sys/build.rs#L222) environment variable, and add `pkg-config` as a build-time dependency.

Reply:
Fixed in [e470d93](https://github.com/Homebrew/homebrew-core/pull/125384/commits/e470d93149a2d817a5ec89fa6c39ac69cb75c79b)
I also confirmed that this formula does indeed link libonig.
Thank you so much.

Linux
```
$ ldd $(which teip)
        linux-vdso.so.1 (0x00007ffcb7f5a000)
        libonig.so.5 => /home/linuxbrew/.linuxbrew/opt/oniguruma/lib/libonig.so.5 (0x00007f8921000000)
        libgcc_s.so.1 => /home/linuxbrew/.linuxbrew/lib/gcc/current/libgcc_s.so.1 (0x00007f892175d000)
        libc.so.6 => /home/linuxbrew/.linuxbrew/opt/glibc/lib/libc.so.6 (0x00007f8920c00000)
        /home/linuxbrew/.linuxbrew/lib/ld.so => /lib64/ld-linux-x86-64.so.2 (0x00007f8921784000)
```

macOS
```
$ ldd $(which teip)
/opt/homebrew/bin/teip:
        /usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
        /opt/homebrew/opt/oniguruma/lib/libonig.5.dylib (compatibility version 9.0.0, current version 9.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
```